### ISSUE TITLE
blue-green deployment plugin failing due to plugin integration issue

### ISSUE THREAD

**Issue by ryanmoran** [cloudfoundry/cli#872]
I am running into an issue with the latest CLI where a plugin that is invoking a command that errors does not receive the error through the plugin API. Specifically, [autopilot](https://github.com/concourse/autopilot) is invoking `cf push` [here](https://github.com/concourse/autopilot/blob/master/autopilot.go#L168). The actually push of the application fails, but no error object is returned, and so `autopilot` thinks everything succeeded.

### Command

The command is for the [autopilot](https://github.com/concourse/autopilot) plugin:
`cf zero-downtime-push -f path/to/manifest.yml -p path/to/code`

### CLI Version

cf version 6.19.0+b29b4e0-2016-06-08

### Error

The plugin doesn't get an error when a command fails.

### Trace

```
$ cf zero-downtime-push notifications -f $PWD/manifest.yml -p $PWD
Renaming app notifications to notifications-venerable in org system / space notifications-service as admin...
OK
Using manifest file manifest.yml

Creating app notifications in org system / space notifications-service as admin...
OK

Using route notifications.bosh-lite.com
Binding notifications.bosh-lite.com to notifications...
OK

Uploading notifications...
Uploading app files from: /var/vcap/data/packages/notifications/10b62a8293b29a836e534647a22d82d3f345e5b5.1-3429e9799cff1750c35bd76fca81f8aebc23c077
Uploading 14.6M, 1549 files
Done uploading
OK

Starting app notifications in org system / space notifications-service as admin...
-----> Downloaded app package (5.8M)
-------> Buildpack version 1.7.8
file:///var/vcap/data/dea_next/admin_buildpacks/4c32f1c6-465b-4fba-87ea-f1ebb368e91b_5dc19602391fb2571b217d754dd7652e6c76630f/dependencies/https___pivotal-buildpacks.s3.amazonaws.com_concourse-binaries_godep_godep-v65-linux-x64.tgz
-----> Checking Godeps/Godeps.json file.
-----> Installing go1.5.4... done
Downloaded [file:///var/vcap/data/dea_next/admin_buildpacks/4c32f1c6-465b-4fba-87ea-f1ebb368e91b_5dc19602391fb2571b217d754dd7652e6c76630f/dependencies/https___storage.googleapis.com_golang_go1.5.4.linux-amd64.tar.gz]
-----> Running: godep go install -v -tags cloudfoundry ./...
github.com/go-gorp/gorp
github.com/go-sql-driver/mysql
github.com/kylelemons/go-gypsy/yaml
github.com/lib/pq/oid
github.com/lib/pq
github.com/mattn/go-sqlite3
github.com/ziutek/mymysql/mysql
github.com/cloudfoundry-incubator/notifications/db
github.com/pivotal-golang/lager
gopkg.in/alexcesaro/quotedprintable.v1/internal
gopkg.in/alexcesaro/quotedprintable.v1
github.com/cloudfoundry-incubator/notifications/metrics
gopkg.in/gomail.v1
github.com/ziutek/mymysql/native
github.com/pivotal-cf-experimental/rainmaker/internal/documents
github.com/pivotal-cf-experimental/rainmaker/internal/network
github.com/cloudfoundry-incubator/notifications/mail
github.com/pivotal-cf-experimental/rainmaker
github.com/dgrijalva/jwt-go
github.com/ziutek/mymysql/godrv
github.com/cloudfoundry-incubator/notifications/cf
github.com/pivotal-cf-experimental/warrant/internal/documents
github.com/pivotal-cf-experimental/warrant/internal/network
github.com/pivotal-cf/uaa-sso-golang/uaa
github.com/pivotal-golang/conceal
github.com/cloudfoundry-incubator/notifications/util
github.com/rubenv/sql-migrate/sqlparse
github.com/pivotal-cf-experimental/warrant
gopkg.in/gorp.v1
github.com/cloudfoundry-incubator/notifications/v2/models
github.com/cloudfoundry-incubator/notifications/uaa
github.com/cloudfoundry-incubator/notifications/v2/collections
github.com/rubenv/sql-migrate
github.com/cloudfoundry-incubator/notifications/postal/common
github.com/cloudfoundry-incubator/notifications/v1/models
github.com/cloudfoundry-incubator/notifications/v2/horde
golang.org/x/net/html/atom
golang.org/x/net/html
github.com/ryanmoran/stack
github.com/cloudfoundry-incubator/notifications/v1/collections
github.com/cloudfoundry-incubator/notifications/v1/web/info
github.com/gorilla/context
github.com/cloudfoundry-incubator/notifications/valiant
github.com/gorilla/mux
github.com/cloudfoundry-incubator/notifications/v2/web/campaigns
github.com/andybalholm/cascadia
github.com/cloudfoundry-incubator/notifications/v1/web/middleware
github.com/cloudfoundry-incubator/notifications/v2/web/campaigntypes
github.com/PuerkitoBio/goquery
github.com/cloudfoundry-incubator/notifications/v2/web/info
github.com/cloudfoundry-incubator/notifications/v2/web/middleware
github.com/cloudfoundry-incubator/notifications/v2/web/root
github.com/cloudfoundry-incubator/notifications/v2/web/senders
github.com/cloudfoundry-incubator/notifications/v2/web/templates
github.com/cloudfoundry-incubator/notifications/v2/web/unsubscribers
github.com/ryanmoran/viron
github.com/cloudfoundry-incubator/notifications/docs
bitbucket.org/liamstask/goose/lib/goose
github.com/cloudfoundry-incubator/notifications/gobble
github.com/cloudfoundry-incubator/notifications/postal/v1
github.com/cloudfoundry-incubator/notifications/v1/services
github.com/cloudfoundry-incubator/notifications/v2/queue
github.com/cloudfoundry-incubator/notifications/v2/web
github.com/cloudfoundry-incubator/notifications/postal/v2
github.com/cloudfoundry-incubator/notifications/v1/web/webutil
github.com/cloudfoundry-incubator/notifications/v1/web/messages
github.com/cloudfoundry-incubator/notifications/v1/web/notify
github.com/cloudfoundry-incubator/notifications/v1/web/clients
github.com/cloudfoundry-incubator/notifications/v1/web/notifications
github.com/cloudfoundry-incubator/notifications/v1/web/preferences
github.com/cloudfoundry-incubator/notifications/v1/web/templates
github.com/cloudfoundry-incubator/notifications/postal
github.com/cloudfoundry-incubator/notifications/v1/web
github.com/cloudfoundry-incubator/notifications/web
github.com/cloudfoundry-incubator/notifications/application
github.com/cloudfoundry-incubator/notifications

-----> Uploading droplet (11M)

0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 starting
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 failing
Deleting app notifications-venerable in org system / space notifications-service as admin...
OK

A new version of your application has successfully been pushed!

Getting apps in org system / space notifications-service as admin...
OK

name            requested state   instances   memory   disk   urls
notifications   started           0/1         64M      1G     notifications.bosh-lite.com
FAILED
```

### Platform Details

```
root@895deb84-703c-40ed-89be-665f8aa0fcbe:~# uname -a
Linux 895deb84-703c-40ed-89be-665f8aa0fcbe 4.2.0-36-generic #42~14.04.1-Ubuntu SMP Fri May 13 17:27:22 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

### Shell

bash

**Comment by XenoPhex** [comment]
We believe that there is a panic/error being raised by push, but the RPC server is not shuffling that error back to the plugin interface. We were able to prove this by inject a `panic` in the middle of the push command and running the `zero-downtime-push` command and we saw it continue to run after the `panic` was triggered with no error being displayed.

We'll try and resolve the `panic` issue, but we have no idea why `push` should be `panic`ing

**Comment by XenoPhex** [comment]
@ryanmoran can you try the [edge build](https://github.com/cloudfoundry/cli/wiki/Edge-Binaries) and see if you guys are getting an error when you push?

**Comment by ryanmoran** [comment]
@XenoPhex I will give this a try and report back.

**Comment by ryanmoran** [comment]
@XenoPhex congrats! It works!

**Comment by dkoper** [comment]
This fix has been released as part of cf CLI 6.20.0.
https://github.com/cloudfoundry/cli#downloads

Thanks for bringing it to our attention!

Regards,
Dies Koper
CF CLI PM
